### ISSUE TITLE
Please create a coverage file, many PRs failed because of this.

### ISSUE THREAD

**Issue by SeasonPanPan** [apache/pulsar#19931]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Example failure

https://github.com/apache/pulsar/actions/runs/4507586495/jobs/7968960455?pr=19906

### Exception stacktrace

```
  Codecov report uploader 0.4.0
[2023-03-26T14:20:49.486Z] ['info'] => Project root located at: /home/runner/work/pulsar/pulsar
[2023-03-26T14:20:49.490Z] ['info'] -> No token specified or token is empty
[2023-03-26T14:20:49.490Z] ['verbose'] Start of network processing...
[2023-03-26T14:20:49.491Z] ['verbose'] Searching for files in /home/runner/work/pulsar/pulsar
[2023-03-26T14:20:49.503Z] ['verbose'] coveragepy is not installed
[2023-03-26T14:20:49.503Z] ['info'] Searching for coverage files...
[2023-03-26T14:20:49.976Z] ['info'] Warning: Some files located via search were excluded from upload.
[2023-03-26T14:20:49.977Z] ['info'] If Codecov did not locate your files, please review https://docs.codecov.com/docs/supported-report-formats
**[2023-03-26T14:20:49.977Z] ['error'] There was an error running the uploader: No coverage files located, please try use `-f`, or change the project root with `-R`**
[2023-03-26T14:20:49.978Z] ['verbose'] The error stack is: Error: No coverage files located, please try use `-f`, or change the project root with `-R`
    at main (/snapshot/repo/dist/src/index.js)
[2023-03-26T14:20:49.978Z] ['verbose'] End of uploader: 501 milliseconds
Error: Codecov: Failed to properly upload: The process '/home/runner/work/_actions/codecov/codecov-action/v3/dist/codecov' failed with exit code 255
```
![image](https://user-images.githubusercontent.com/25130182/227785298-ac3f9add-97a1-4ed1-a92c-861ceff1696d.png)

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by tisonkun** [comment]
Cross refer to https://lists.apache.org/thread/4kjp5bdct282z6o11jvjvlxkjgmcc4ms

**Comment by tisonkun** [comment]
Upstream issue - https://community.codecov.com/t/upload-issues-unable-to-locate-build-via-github-actions-api/3954

@lhotari @BewareMyPower @yaalsn shall we remove/disable the codecov step for now?

**Comment by BewareMyPower** [comment]
My personal answer is to remove it.

**Comment by tisonkun** [comment]
It seems that we don't create the coverage file. It may be a workflow bug.

**Comment by yaalsn** [comment]
Could we use `continue-on-error` for the flaky test?

**Comment by tisonkun** [comment]
@yaalsn I guess the following steps in the current workflow is already run `always`:

https://github.com/apache/pulsar/blob/d14e43ed9bd079b76301f10dc1c039cafd43e1cf/.github/workflows/pulsar-ci-flaky.yaml#L168-L196

**Comment by lhotari** [comment]
> It seems that we don't create the coverage file. It may be a workflow bug.

@tisonkun that's right. there seems to be an issue in that.

**Comment by lhotari** [comment]
@tisonkun I'm currently experimenting a solution which would be more reliable in dumping the Jacoco coverage data to files, WIP commit https://github.com/lhotari/pulsar/pull/144/commits/5bd1b74a90bcc6e9d294328ad74e068133319380 . I debugged the problem and the Jacoco execution files were empty. One possible reason for that is that another shutdown hook blocks the shutdown sequence. I've worked around that possible problem by using the JMX API for dumping the Jacoco exec data in a TestNG listener.
